processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByMergingQueryRunnerV2.java

```diff
@@ -192,40 +170,39 @@ public GroupByMergingQueryRunnerV2(
                   ReferenceCountingResourceHolder.fromCloseable(temporaryStorage);
               resources.add(temporaryStorageHolder);
 
-              final ReferenceCountingResourceHolder<ByteBuffer> mergeBufferHolder;
-              try {
-                // This will potentially block if there are no merge buffers left in the pool.
-                if (hasTimeout) {
-                  final long timeout = timeoutAt - System.currentTimeMillis();
-                  if (timeout <= 0 || (mergeBufferHolder = mergeBufferPool.take(timeout)) == null) {
-                    throw new TimeoutException();
-                  }
-                } else {
-                  mergeBufferHolder = mergeBufferPool.take();
-                }
-                resources.add(mergeBufferHolder);
-              }
-              catch (Exception e) {
-                throw new QueryInterruptedException(e);
-              }
+              // If parallelCombine is enabled, we need two merge buffers for parallel aggregating and parallel combining
+              final int numMergeBuffers = querySpecificConfig.getNumParallelCombineThreads() > 1 ? 2 : 1;
```

Comment:
What if 2 merge buffers are not available? For example if numMergeBuffers is set to 1.

Also, the config docs should be updated to reflect that when parallel combining is used, the number of merge buffers needed will double.

Reply:
`getMergeBuffer()` takes required merge buffers atomically. If required buffers are not available, it would throw an exception. I added a check for the size of mergeBufferPool and changed to throw proper exceptions.

Also updated doc.